src/hooks/connectWeb3/useWeb3React.tsx

```diff
@@ -70,30 +70,32 @@ interface ActiveWeb3ProviderProps {
   defaultChainId: SupportedChainId
 }
 
+const onError = console.error
```

Comment:
if for whatever reason we're worried about `console.error` not being defined at file load time we could move this into a `useRef` in the function component body, but don't think that should be done unless we have an explicit reason to

Reply:
This prevents developers from monkey-patching console.error. I think it's preferable to invoke it lazily, and not cache it on file load.

What's wrong with `const onError = console.error` in the `ActiveWeb3Provider` fn body?

Reply:
my worry was that console.error might be changing identity every render, which would have caused a lot of the memos to fire again without changing the key

your point about monkey-patching is good though, and also my second commit should fix the underlying issue even if console.error is changing all the time, so i'll revert this bit